<launch>
    <arg name="robot_name"/>
    <arg name="test_mode"/>

    <arg name="width"       default="320"/>
    <arg name="height"      default="240"/>
    <arg name="fps"         default="10"/>
    <!-- Add an argument for camera_info_url with a default value -->
    <arg name="camera_info_url" default="file://$(find vpa_robot)/config/camera_default.yaml"/>
    
    <node name="robot_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
        <param name="video_device"                  value="/dev/video0" />
        <param name="image_width"                   value="$(arg width)" />
        <param name="image_height"                  value="$(arg height)" />
        <param name="pixel_format"                  value="yuyv" />
        <!-- Use the configurable argument for camera_info_url -->
        <param name="camera_info_url"               value="$(arg camera_info_url)"/> 
        <param name="camera_frame_id"               value="base_camera_link" />
        <param name="framerate"                     value="$(arg fps)"/>
        <param name="io_method"                     value="mmap"/>
    </node>

    <node name="robot_vision" pkg="vpa_robot" type="visual_detector.py" output="screen">
        <param name="robot_name"    value="$(arg robot_name)"/>
        <param name="image_width"   value="$(arg width)"/>
        <param name="image_height"  value="$(arg height)"/>
        <param name="test_mode"     value="$(arg test_mode)"/>
        <!-- lane detecting - yellow lane -->
        <param name="h_upper_1"            value="36"/>
        <param name="h_lower_1"            value="26"/>
        <param name="s_upper_1"            value="255"/>
        <param name="s_lower_1"            value="100"/>
        <param name="v_upper_1"            value="255"/>
        <param name="v_lower_1"            value="100"/>
        <!-- lane detecting - white lane -->
        <param name="h_upper_2"            value="180"/>
        <param name="h_lower_2"            value="0"/>
        <param name="s_upper_2"            value="70"/>
        <param name="s_lower_2"            value="0"/>
        <param name="v_upper_2"            value="255"/>
        <param name="v_lower_2"            value="200"/>
        <!-- stop line (red)-->
        <param name="h_upper_s"            value="8"/>
        <param name="h_lower_s"            value="0"/>
        <param name="s_upper_s"            value="255"/>
        <param name="s_lower_s"            value="150"/>
        <param name="v_upper_s"            value="255"/>
        <param name="v_lower_s"            value="150"/>
    </node>

    <node name="v4l2_setup_node" pkg="vpa_robot_vision" type="setup_v4l2.py" output="screen" />
</launch>